packages/gatsby-source-shopify/src/query-builders/bulk-query.ts

```diff
@@ -3,6 +3,13 @@ export abstract class BulkQuery {
 
   constructor(pluginOptions: ShopifyPluginOptions) {
     this.pluginOptions = pluginOptions
+
+    if (
+      process.env.GATSBY_SHOPIFY_SALES_CHANNEL &&
+      !this.pluginOptions.salesChannel
+    ) {
+      this.pluginOptions.salesChannel = process.env.GATSBY_SHOPIFY_SALES_CHANNEL
```

Comment:
We probably shouldn't do it this way

Reply:
Maybe it would work to modify the `pluginOptionsSchema` function

```ts
salesChannel: Joi.string().default(process.env.GATSBY_SHOPIFY_SALES_CHANNEL)
```